res/res_pjsip_outbound_registration.c

```diff
@@ -743,6 +750,27 @@ static pj_status_t registration_client_send(struct sip_outbound_registration_cli
 	/* Add Security-Verify or Security-Client headers */
 	add_security_headers(client_state, tdata);
 
+	/*
+	 * Replace the User-Agent-header if a different one should be used
+	 */
+	if (!ast_strlen_zero(client_state->user_agent)) {
+		static const pj_str_t user_agent_str = { "User-Agent", 10 };
+		pjsip_generic_string_hdr *default_user_agent_hdr;
+		pjsip_generic_string_hdr *user_agent_hdr;
+		pj_str_t user_agent_val;
+		default_user_agent_hdr = pjsip_msg_find_hdr_by_name(tdata->msg, &user_agent_str, NULL);
+		user_agent_val = pj_str(client_state->user_agent);
+		user_agent_hdr = pjsip_generic_string_hdr_create(tdata->pool, &user_agent_str, &user_agent_val);
+		if (!user_agent_hdr) {
+			ast_log(LOG_ERROR, "Could not add custom User-Agent to outbound registration %s, sending REGISTER request with non-custom header\n", client_state->registration_name);
+		} else {
+			if (default_user_agent_hdr) {
+				pj_list_erase(user_agent_hdr);
```

Comment:
Shouldn't this be `pj_list_erase(default_user_agent_hdr);`

Reply:
Yes indeed. Fixed.